src/app/editor/SourceHighlighters.js

```diff
@@ -1,24 +1,22 @@
 'use strict'
 const SourceHighlighter = require('./sourceHighlighter')
 
-import { ApiFactory } from 'remix-plugin'
+import { EditorApi } from 'remix-plugin'
 
-class SourceHighlighters extends ApiFactory {
+const profile = {
+  displayName: 'source highlighters',
```

Comment:
`displayName` and `name` are not necessary right? they will be overridden by the `super` of `EditorApi`

Reply:
No, only the methods, events and kind are going to be overrided.

Reply:
oh ok, then I think we should have a default naming for API shared among different platform. That does not need to be completely typescripted but at least well documented.
i.e if a file system is usually accessible under `call('fileSystem', '<function>', [])` , others platform have to provide it under the same name `fileSystem`.

in a further enhancement, we could add a function for retrieving modules of a specific `kind`.

Reply:
Sounds fair, but how you would deal with `compiler` ? You'll have the same `kind` with the same `methods` and `events`, but the names are different (`solidity`, `vyper`, `lll`, ...).

Reply:
That would only apply with the core API like file manager, editor, context, udapp

Reply:
That would only apply with the core API like file manager, editor, context, udapp

Reply:
That would only apply with the core API like file manager, editor, context, udapp

Reply:
So the core API would be the one that are unique ?
Like : 
- core : `fileSystem`, `editor`, `network`, `udapp`.
- extension : `compiler`, `test`, `analysis`, `vm` (?)

Reply:
Why 'vm' ? 
Yes, The core API provides infrastructure for module/plugin to interact with.

Reply:
Well, I guess at some point we would like to support eWASM Virtual Machine for example.